detect_secrets/plugins/core/constants.py

```diff
@@ -11,7 +11,18 @@
             ('\'', ''),             # e.g. visual basic .net
             ('--', ''),             # e.g. sql
             # many other inline comment syntaxes are not included,
-            # because we want to be performant for the common case
+            # because we want to be performant for
+            # any(regex.search(line) for regex in WHITELIST_REGEXES)
+            # calls. of course, this won't be a concern if detect-secrets
+            # switches over to implementing file plugins for each supported
+            # filetype.
         )
     ]
 ]
+
+# add to this mapping (and WHITELIST_REGEXES if applicable) lazily,
+# as more language specific file parsers are implemented.
+# discussion: https://github.com/Yelp/detect-secrets/pull/105
+WHITELIST_REGEX = {
+    'yaml': WHITELIST_REGEXES[0],
```

Comment:
Super Nit: Maybe add to the regex comment `# e.g. python` -> `# e.g. python or yaml`

Reply:
I don't think that'll be necessary if a python file parser is implemented. Then we can `'py': WHITELIST_REGEXES[0]`. More language/specification specific file parsers is what I'm pushing for, or at least suggesting, architecturally. See #105.